controller/src/main/java/io/pravega/controller/store/stream/tables/State.java

```diff
@@ -9,12 +9,39 @@
  */
 package io.pravega.controller.store.stream.tables;
 
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
 /**
```

Comment:
It will still act as a lock even when stream creation is moved to event processors. 
this is because we need some 'state' that prevents concurrent conflicting update to metadata from happening.